autosubmit/job/job_list.py

```diff
@@ -1417,20 +1436,33 @@ def _manage_job_dependencies(self, dic_jobs, job, date_list, member_list, chunk_
                                             key_aux_stripped.split("+")[0] if "+" in key_aux_stripped else
                                             key_aux_stripped for key_aux_stripped in aux.keys()]
 
-            problematic_dependencies = self._calculate_natural_dependencies(dic_jobs, job, dependency, date,
-                                                                            member, chunk, graph,
-                                                                            distances_of_current_section, key,
-                                                                            dependencies_of_that_section, chunk_list,
-                                                                            date_list, member_list,
-                                                                            special_dependencies, max_distance,
-                                                                            problematic_dependencies)
+            problematic_dependencies = self._calculate_natural_dependencies(
+                dic_jobs, job, dependency, date, member, chunk, graph,
+                distances_of_current_section, key,
+                dependencies_of_that_section, chunk_list,
+                date_list, member_list,
+                special_dependencies, max_distance,
+                problematic_dependencies
+            )
 
         return problematic_dependencies
 
-
     @staticmethod
-    def _calculate_dependency_metadata(chunk, chunk_list, member, member_list, date,
-                                       date_list, dependency):
+    def _calculate_dependency_metadata(
+            chunk: Optional[int],
+            chunk_list: List[int],
+            member: Optional[str],
+            member_list: List[str],
+            date: Optional[str],
+            date_list: List[str],
+            dependency: Any
+    ) -> Tuple[bool, Tuple[Optional[int], Optional[str], Optional[str]]]:
+        """
+        Compute the target chunk/member/date for a dependency with a +/- distance.
```

Comment:
Would be good if we started documenting the code following PEP 257, where the first comment is a very short note about the code. Followed by a blank line, then the body of the comment.

It's actually parsed that way, with a title for the comment, followed by the body. And then args/returns/etc. using Google/napoleon'etc. styles.

But this

```
"""
Bla.
"""
```

is not the same as 

```
"""Bla."""
```

It's similar to the example in this part, https://peps.python.org/pep-0257/#handling-docstring-indentation

> The docstring in this example contains two newline characters and is therefore 3 lines long. The first and last lines are blank

It's harmless, but eventually we might want to have it the way of that PEP for better integration with tools like Sphinx, and also for consistency, I think.